### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by badhrinathpa**
1. Reset current proc after attach and service req procedures.
2. Fix Ctx release command criticality param. ENB was ignoring the packet.
3. UE flush using Monitoring Tool. (Test Feature)
4. Add Ue State in Monitoring Tool info.
4. Set UE to Idle state during Context Release.

### CODE REVIEW

File: src/mme-app/handlers/s1ap_common_req_handler.c [hunk 1]
```diff
@@ -129,7 +129,22 @@ ctx_rel_req_processing()
...
+    char imsiStr[16] = {0};
+    imsi_bin_to_str(ue_entry->IMSI, imsiStr);
+    switch(ue_entry->ue_state)
+    {
+        case DETACH_STAGE1:
+        case DETACH_STAGE2_DS_DONE:
+        case DETACH_STAGE2_PURGE_DONE:
+        case DETACH_STAGE2:
+            log_msg(LOG_DEBUG, "Ctx Release during Detach for IMSI. Skip RABR: %s\n", imsiStr);
+            return E_FAIL;
```
Comment (thakurajayL) [inline src/mme-app/handlers/s1ap_common_req_handler.c]: Does not look like failure. Its perfectly normal behavior right ?

File: src/mme-app/monitor_subscriber.c [hunk 2]
```diff
@@ -114,6 +116,24 @@ handle_imsi_list_req(struct monitor_imsi_req *mir, int sock_fd)
...
+* Flush Imsis.
+*/
+void
+handle_imsi_flush_req()
+{
+    struct UE_info *head_entry = ue_list_head;
+    log_msg(LOG_DEBUG, "imsi flush request, ue count %d\n", g_total_UE_count);
+    while(head_entry)
+    {
+        release_ue_entry(head_entry);
```
Comment (thakurajayL) [inline src/mme-app/monitor_subscriber.c:129]: Can we send message to UE to reattach..That would really be good in testing

File: src/mme-app/monitor_subscriber.c [hunk 3]
```diff
@@ -139,7 +159,12 @@ handle_monitor_processing(void *message)
 	            handle_imsi_list_req(mir, sock_fd);
                 break;
             }
-            
+        case 2:
```
Comment (thakurajayL) [inline src/mme-app/monitor_subscriber.c:162]: how do you send this case 2 ?

File: include/mme-app/ue_table.h [hunk 4]
```diff
@@ -66,6 +66,7 @@ enum ue_stages{
   DETACH_STAGE2_DS_DONE,
   DETACH_STAGE2,
   DETACH_DONE,
+  CTX_RELEASE_STAGE,
```
Comment (thakurajayL) [inline include/mme-app/ue_table.h]: This s critical.  Lets add enough information about what are these states mean.

File: include/mme-app/ue_table.h [hunk 5]
```diff
@@ -85,6 +86,7 @@ enum ue_proc{
 };
 
 enum ecm_states{
+ ECM_UNKNOWN=0,
```
Comment (thakurajayL) [inline include/mme-app/ue_table.h:92]: Add a comment which says when we move to IDLE state and when we move to connected state.

File: src/mme-app/handlers/attach_aia_ula.c [hunk 6]
```diff
@@ -294,7 +294,7 @@ post_to_next(int ue_index)
 	}
     else if (STAGE1_AIA_FAIL == ue_entry->ue_state)
     {
-        log_msg(LOG_ERROR, "Error AIA from HSS. Release UE session \n");
+        log_msg(LOG_ERROR, "Error AIA from HSS. Send Attach Reject & Release UE session \n");
```
Comment (thakurajayL) [inline src/mme-app/handlers/attach_aia_ula.c]: Print IMSI. We should slowly target the log cleanup which helps in debugging issues without restarting MME pod

File: src/mme-app/handlers/s1ap_common_req_handler.c [hunk 7]
```diff
@@ -129,7 +129,22 @@ ctx_rel_req_processing()
...
 
+    char imsiStr[16] = {0};
+    imsi_bin_to_str(ue_entry->IMSI, imsiStr);
+    switch(ue_entry->ue_state)
+    {
+        case DETACH_STAGE1:
+        case DETACH_STAGE2_DS_DONE:
+        case DETACH_STAGE2_PURGE_DONE:
+        case DETACH_STAGE2:
+            log_msg(LOG_DEBUG, "Ctx Release during Detach for IMSI. Skip RABR: %s\n", imsiStr);
```
Comment (thakurajayL) [inline src/mme-app/handlers/s1ap_common_req_handler.c:140]: Can you please check how this is handled in new MME..and open issue if new MME is not handling this case ?

**Review by thakurajayL — CHANGES_REQUESTED** [review]
1) Please open issues if some requested code is missing from new MME.
2) add comments in openmme code

File: src/mme-app/monitor_subscriber.c (hunk 3 above)
Comment (badhrinathpa) [reply src/mme-app/monitor_subscriber.c:162]: We send the type in exporter python file. Not added to that file yet. Can be added when needed.

File: src/mme-app/handlers/s1ap_common_req_handler.c (hunk 7 above)
Comment (badhrinathpa) [reply src/mme-app/handlers/s1ap_common_req_handler.c:140]: Not handled in Nucleus. Will open Issue.

File: src/mme-app/monitor_subscriber.c (hunk 2 above)
Comment (badhrinathpa) [reply src/mme-app/monitor_subscriber.c:129]: Will raise issue for this update.

File: src/mme-app/handlers/s1ap_common_req_handler.c (hunk 1 above)
Comment (badhrinathpa) [reply src/mme-app/handlers/s1ap_common_req_handler.c]: We have to skip RABR. Success will send RABR. Will update to E_SKIP.

File: include/mme-app/ue_table.h (hunk 5 above)
Comment (badhrinathpa) [reply include/mme-app/ue_table.h:92]: done.

File: include/mme-app/ue_table.h (hunk 4 above)
Comment (badhrinathpa) [reply include/mme-app/ue_table.h]: done.

**Pull request merged**